fix(tagstore): Always pass a dataset in tagstore

wmak opened the pull request:
- Most of these are group specific, so I'm using Dataset.Events for
  those
- get_tag_keys_for_projects, is used across the app for autocomplete,
  and getting a list of columns. So it uses the Discover dataset by
  default now, instead of Events
- Will test this in staging first once I get approval before merging since a lot of pages will get impacted by this.

wmak committed:
fix(tagstore): Always pass a dataset in tagstore

- Most of these are group specific, so I'm using Dataset.Events for
  those
- get_tag_keys_for_projects, is used across the app for autocomplete,
  and getting a list of columns. So it uses the Discover dataset by
  default now, instead of Events

wmak commented on src/sentry/tagstore/snuba/backend.py:
```diff
@@ -333,6 +342,7 @@ def get_tag_keys_for_projects(
             end,
             include_values_seen=False,
             use_cache=use_cache,
+            dataset=Dataset.Discover,
```
Because we currently default to the Events dataset in https://github.com/getsentry/sentry/blob/master/src/sentry/utils/snuba.py#L594 this is really the only change.
Added events everywhere else just so that we start being explicit with the dataset being used

lynnagara reviewed (changes requested):
I think it's pretty unexpected for the default dataset for some tagstore functions to be different than others. I think it'd be better to always apply events as a default everywhere and change it specifically where this is needed like the other method in this module that needs to use Discover does.

Also some methods now use `include_transactions` boolean to switch to Discover, and some include a `dataset` argument to achieve the same thing. Let's make this consistent.

wmak committed:
ref: use include_transactions to be more consistent

- Defaults to false, but defaulting it to true for the organization_tags
  endpoint since the frontend caches this and it gets re-used between
  issues and discover

wmak committed:
Merge branch 'master' into wmak/fix/tagstore-dataset
